component/attr_cache/attr_cache.go

```diff
@@ -766,9 +682,12 @@ func (ac *AttrCache) TruncateFile(options internal.TruncateFileOptions) error {
 		ac.cacheLock.RLock()
 		defer ac.cacheLock.RUnlock()
 
-		// no need to truncate the name of the file
-		value, found := ac.cacheMap[options.Name]
-		if found && value.valid() && value.exists() {
+		value, getErr := ac.cacheMap.get(options.Name)
+		if getErr != nil {
+			log.Err("AttrCache::TruncateFile : could not truncate file due to the following error: %s", getErr)
+			return getErr
+		}
+		if value.exists() {
 			value.setSize(options.Size)
 		}
```

Comment:
```suggestion
		truncatedItem, getErr := ac.cacheMap.get(options.Name)
		if getErr != nil || !truncatedItem.exists() {
            log.Warn("AttrCache::TruncateFile : %s replacing missing cache entry", options.Name)
            // replace the missing entry
            entryTime := time.Now()
            truncatedAttr := internal.CreateObjAttr(options.Name, options.Size, entryTime)
			truncatedItem = ac.cacheMap.insert(truncatedAttr, true, entryTime)
		}
        truncatedItem.setSize(options.Size)
```
--If there is no entry, one can be created.
--Simplify log message
--No need to return the error from get() in any case, since the TruncateFile call to the cloud had no error.